### DISCUSSION TITLE
I ran the smart contract and 127.0.1.1.7545 on my device, and I linked them with the Python file on the virtual box, when running them it gives me the previous error.

### DISCUSSION THREAD

**Discussion by thakuradi** [smartcontractkit/full-blockchain-solidity-course-py#1777] [Q&A] [answered]
raise ConnectionError(e, request=request)
requests. exceptions.ConnectionError: HTTPConnectionPool(host='0.0.0.0', port=7545): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f24fd35ef10>: Failed to establish a new connection: [Errno 111] Connection refused'))

this is the error I am getting. how to fix even I do each step carefully.

**Comment by cromewar** [comment, marked as answer]
Hello @thakuradi 
If you are using virtual box it means the local host configuration is not the same, go to virtual box network configuration and see the correct ip address.

If you meant you are using python virtualenv (not the same as virtualbox), you should try using other than 0.0.0.0 try `127.0.0.1` instead.

I hope this info might help, cheers.

**Reply by thakuradi** [reply]
I am using WSL for Linux in my VSC, I tried the 2nd method still getting the error please help me.

**Reply by cromewar** [reply]
Oh if it is WSL, you should go to your ganache UI configuration and change the host server to WSL option, this will change the ip address to a new one and you should use that one instead.

**Reply by thakuradi** [reply]
very very thanks bro , really i am trying to fix that from 2dayss , thanks again

**Reply by cromewar** [reply]
did it solve the problem?
